core/src/blocktree.rs

```diff
@@ -193,66 +193,69 @@ impl Blocktree {
         false
     }
 
-    // silently deletes all blocktree column families starting at the given slot
-    fn delete_all_columns(&self, starting_slot: u64) {
-        match self.meta_cf.force_delete_all(Some(starting_slot)) {
+    /// Silently deletes all blocktree column families starting at the given slot until the `to` slot
+    /// Use with care; does not check for integrity and does not update slot metas that
+    /// refer to deleted slots
+    pub fn delete_all_columns(&self, from_slot: u64, to_slot: Option<u64>) {
```

Comment:
can you pass an std::ops::Range to this function?

Reply:
I tried using rangebounds (so the we can do `(n.. m)`) index doesn't support a default (to start at 0)and it basically became really hacky. A std::ops::Range _requires_ a lower and upper bound.

I can give it another shot but it requires index to turn into a more complex type that every cf will have to implement.

Reply:
not worth it